src/plugins/discover/public/application/apps/main/utils/get_switch_index_pattern_app_state.ts

```diff
@@ -38,9 +38,8 @@ export function getSwitchIndexPatternAppState(
   return {
     index: nextIndexPattern.id,
     columns,
-    sort:
-      nextIndexPattern.timeFieldName && !nextSort.length
-        ? [[nextIndexPattern.timeFieldName, sortDirection]]
-        : nextSort,
+    sort: nextIndexPattern.timeFieldName
+      ? [[nextIndexPattern.timeFieldName, sortDirection]]
+      : nextSort,
```

Comment:
didn't test but I think this would mean, that we would switch to sort by the timeFieldName of the next index pattern in any case. this would resolve the issue. but it would also open another one. 
Let's say you have sorted by `_score` and you would switch index pattern then. You'd loose the current sort then. So I think a check is necessary if the current sort contains the current `timeFieldName`. if this is the case it makes sense to switch to sort by the `timeFieldName` of the next index pattern.

Reply:
Thx for the findings. Now prev sorting shouldn't be lost.